scipy/conftest.py

```diff
@@ -206,7 +193,21 @@ def num_parallel_threads():
     from cupyx.scipy import signal
     del signal
 
-array_api_compatible = pytest.mark.parametrize("xp", xp_available_backends.values())
+
+@pytest.fixture(params=[
+    pytest.param(v, id=k, marks=pytest.mark.array_api_backends)
+    for k, v in xp_available_backends.items()
+])
+def xp(request):
+    if SCIPY_ARRAY_API:
+        from scipy._lib._array_api import default_xp
+
+        with default_xp(request.param):
+            yield request.param
+    else:
+        yield request.param
```

Comment:
This fixture, and `default_xp` needs code comments. What does it actually do and how is it activated?